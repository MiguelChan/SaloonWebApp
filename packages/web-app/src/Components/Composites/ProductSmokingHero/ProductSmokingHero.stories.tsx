import React from 'react';
import {
  Story,
  Meta,
} from '@storybook/react';
import ProductSmokingHero from './ProductSmokingHero';

export default {
  title: 'Components/Composites/ProductSmokingHero',
  component: ProductSmokingHero,
} as Meta;

const Template: Story = () => <ProductSmokingHero />;

export const Default = Template.bind(this);