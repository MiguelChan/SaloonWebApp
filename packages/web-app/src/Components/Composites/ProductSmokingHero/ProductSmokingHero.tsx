import * as React from 'react';
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import Container from '@mui/material/Container';
import { Typography } from 'Components/Blocks';
import helpWanted from 'Images/helpWanted.svg';

const ProductSmokingHero: React.FunctionComponent = () => {

  return (
    <Container
      component="section"
      sx={{ display: 'flex', flexDirection: 'column', alignItems: 'center', my: 9 }}
    >
      <Button
        sx={{
          border: '4px solid currentColor',
          borderRadius: 0,
          height: 'auto',
          py: 2,
          px: 5,
        }}
        target='_blank'
        component='a'
        href='https://wa.me/5219996035008'
      >
        <Typography variant="h4" component="span">
          Tienes alguna duda? Necesitas ayuda?
        </Typography>
      </Button>
      <Typography variant="subtitle1" sx={{ my: 3 }}>
        Estamos aquí para ayudarte, !Contáctanos!
      </Typography>
      <Box
        component="img"
        src={helpWanted}
        alt="buoy"
        sx={{ width: 60 }}
      />
    </Container>
  );
}

export default ProductSmokingHero;