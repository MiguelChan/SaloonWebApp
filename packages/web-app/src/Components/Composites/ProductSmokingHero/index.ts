export { default as ProductSmokingHero } from './ProductSmokingHero';
export * from './ProductSmokingHero';