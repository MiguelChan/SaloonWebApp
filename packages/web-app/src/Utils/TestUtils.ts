import { ProductStep, ProductValue } from "@mchanc/shared-library";

export const renderProductValues = (numberOfElements: number): ProductValue[] => {
  const productValues: ProductValue[] = [];
  for (let i = 0; i < numberOfElements; i++) {
    const productValue = {
      imageUrl: 'https://source.unsplash.com/random',
      altTitle: `Test + ${i}`,
      title: `Another Title: ${i}`,
      description: `This is some description for ${i}`,
    };
    productValues.push(productValue);
  }
  return productValues;
}

export const renderProductSteps = (numberOfElements: number): ProductStep[] => {
  const productSteps: ProductStep[] = [];
  for (let i = 0; i < numberOfElements; i++) {
    const productStep: ProductStep = {
      stepImage: 'https://source.unsplash.com/random',
      stepNumber: i,
      stepDescription: `This is some description for ${i}`,
    };
    productSteps.push(productStep);
  }
  return productSteps;
}