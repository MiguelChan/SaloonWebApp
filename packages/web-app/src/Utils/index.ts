export * from './Validation';
export { default as theme } from './Theme';