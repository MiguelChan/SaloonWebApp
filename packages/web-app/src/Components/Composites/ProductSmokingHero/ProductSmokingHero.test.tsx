import React from 'react';
import {
  render,
} from '@testing-library/react';
import ProductSmokingHero from './ProductSmokingHero';
import { ThemeProvider } from '@mui/styles';
import { theme } from 'Utils';

describe('ProductSmokingHero', () => {

  it('Should match snapshot', () => {
    const {
      container,
    } = setupComponent();

    expect(container).toMatchSnapshot();
  });

  const setupComponent = () => {
    const Component = (
      <ThemeProvider theme={theme}>
        <ProductSmokingHero />
      </ThemeProvider>
    );

    return render(Component);
  };
});