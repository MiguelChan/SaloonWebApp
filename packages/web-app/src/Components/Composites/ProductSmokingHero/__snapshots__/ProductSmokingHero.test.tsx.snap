// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductSmokingHero Should match snapshot 1`] = `
<div>
  <section
    class="MuiContainer-root MuiContainer-maxWidthLg css-14iqigo-MuiContainer-root"
  >
    <a
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-10c5xe0-MuiButtonBase-root-MuiButton-root"
      href="https://wa.me/5219996035008"
      tabindex="0"
      target="_blank"
    >
      <span
        class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
      >
        Tienes alguna duda? Necesitas ayuda?
      </span>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
    <h3
      class="MuiTypography-root MuiTypography-subtitle1 css-1t5ttap-MuiTypography-root"
    >
      Estamos aquí para ayudarte, !Contáctanos!
    </h3>
    <img
      alt="buoy"
      class="MuiBox-root css-q2ms4q"
      src="helpWanted.svg"
    />
  </section>
</div>
`;
